### PR TITLE
Autotest: Python fixes

### PR DESCRIPTION
**Pull request by khancyr**
Some python fixes for autotest.
One fix solve a compatibility issue for python3
I am unsure about what intend for bad char handling on FRSKYSPort

### CODE REVIEW

File: Tools/autotest/common.py [hunk 1]
```diff
@@ -456,8 +458,8 @@ def do_sport_read(self):
                 if b != self.SPORT_DATA_FRAME:
                     # we may come into a stream mid-way, so we can't judge
                     self.progress("############# Bad char %x" % b)
-                    raise ValueError("Bad char (0x%02x)" % b)
```
Comment (peterbarker) [inline Tools/autotest/common.py]: Eep!  major function change.  Please revert this one and fix whatever needed fixing in some other way :-)

Removing bad_chars entirely is an option.

**Review by peterbarker — CHANGES_REQUESTED** [review]
Some good changes in here.

File: Tools/autotest/common.py (hunk 1 above)
Comment (khancyr) [reply Tools/autotest/common.py]: Ok so raising the exception was the good way?
I will revert and do the same for the other case

**Comment by khancyr** [comment]
@peterbarker ping

**Comment by peterbarker** [comment]
Squashed it all together and merged it - thanks!

**Pull request closed without merging**